SYCL/ESIMD/slm_split_barrier.cpp

```diff
@@ -0,0 +1,166 @@
+//==----------- slm_split_barrier.cpp  - DPC++ ESIMD on-device test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows
+// REQUIRES: linux && gpu
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out
+
```

Comment:
`esimd_sbarrier` is not available yet AFAICT, so this will fail.

Reply:
This support is included in another pending PR #2917 to GitHub.